spec/unit/resource/timezone_spec.rb

```diff
@@ -36,4 +61,42 @@
     expect { resource.action :set }.not_to raise_error
     expect { resource.action :unset }.to raise_error(Chef::Exceptions::ValidationFailed)
   end
+
+  describe "#current_darwin_tz?" do
+    context "with admin privs" do
+      it "returns the TZ" do
+        expect(resource).to receive(:shell_out!).and_return(shellout_systemsetup)
+        expect(resource.current_darwin_tz).to eql("UTC")
+      end
+    end
+
+    context "without admin privs" do
+      it "returns the TZ" do
+        expect(resource).to receive(:shell_out!).and_return(shellout_systemsetup_fail)
+        expect { resource.current_darwin_tz }.to raise_error(RuntimeError)
```

Comment:
Given RuntimeError is pretty generic this could also check for some portion of the error message.

Reply:
Fixed